### PR TITLE
Bump default dbcache

### PR DESCRIPTION
**Pull request by dagurval**
Cherry picks of https://github.com/bitcoin/bitcoin/pull/8273 amd https://github.com/bitcoin/bitcoin/pull/8407

### CODE REVIEW

File: src/qt/optionsmodel.cpp [hunk 1]
```diff
@@ -475,3 +477,22 @@ bool OptionsModel::isRestartRequired()
...
+    // Check if the QSettings container was already loaded with this client version
+    QSettings settings;
+    static const char strSettingsVersionKey[] = "nSettingsVersion";
+    int settingsVersion = settings.contains(strSettingsVersionKey) ? settings.value(strSettingsVersionKey).toInt() : 0;
+    if (settingsVersion < CLIENT_VERSION)
+    {
+        // -dbcache was bumped from 100 to 300 in 0.13
+        // see https://github.com/bitcoin/bitcoin/pull/8273
+        // force people to upgrade to the new value if they are using 100MB
+        if (settingsVersion < 130000 && settings.contains("nDatabaseCache") && settings.value("nDatabaseCache").toLongLong() == 100)
```
Comment (dagurval) [inline src/qt/optionsmodel.cpp]: Right.

CLIENT_VERSION_XT_SUBVER is not part of CLIENT_VERSION, so this is not compatible with XT versioning. How about we leave out this commit?

**Comment by dgenr8** [comment]
@dagurval Were you going to remove the second commit?

**Comment by dagurval** [comment]
Done

**Comment by dgenr8** [comment]
utACK

**Pull request merged**